python/ray/data/datasource/image_datasource.py

```diff
@@ -75,6 +77,9 @@ def _read_stream(
     ) -> Iterator[Block]:
         from PIL import Image, UnidentifiedImageError
 
+        if self.max_image_pixels is not None:
+            Image.MAX_IMAGE_PIXELS = self.max_image_pixels
```

Comment:
this updates the the global variable, and will also affect the next `read_images` call. 
Is there a way to avoid this?